### PR TITLE
feat(codeowners): add code reviewers to the repository

### LINKED ISSUE #3391: Add code owners and turn on automatic reviewer assignment
## Motivation

In this week's Engineering sync, we decided to turn on code owners and automatic reviewer assignments.

## Reference Documentation

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team

## Code Owner Lists

These parts of the codebase should be assigned to specific reviewers.

### Network and Async Code

To be reviewed by Teor, Janito, or Alfredo.

```
tower-batch
tower-fallback
zebra-consensus
    except for zebra-consensus/src/primitives, which is cryptographic
zebra-network
zebra-state
zebra-tests/src/mock_service.rs
zebra-tests/src/service_extensions.rs
zebra-tests/src/transcript.rs
zebra-utils
zebrad/src/commands
zebrad/src/components
```

### Cryptographic Code

To be reviewed by Deirdre, Conrado, or Marek.

```
zebra-consensus/src/primitives
zebra-chain/src/primitives
zebra-chain/src/orchard
zebra-chain/src/sapling
zebra-chain/src/sprout
zebra-chain/src/transparent
zebra-chain/src/history_tree.rs
zebra-chain/src/history_tree
```

### Devops Code

To be reviewed by Gustavo, Deirdre, Teor, or Conrado.

```
.github
docker
cloudbuild.yaml
codecov.yml 
```

### Unsafe Code

To be reviewed by Teor, Janito, Conrado, or Marek.

We should automatically assign 2 reviewers to changes to this code. (If that's possible using GitHub or Mergify.)

```
zebra-script
```

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

Code owners are automatically requested for review when someone opens a pull request that modifies code that they own.

## Solution
- Create teams with distribution based on modules/responsibility they have on the code.
- Create a CODEOWNERS file with the owners definition

Closes #3391 

## Review

@teor2345 can review this

## Follow Up Work
The teams I created need write permission, and should preferably be under the ZFND team. But I don't have enough permissions to edit the teams with this information.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# review when someone opens a pull request.
+*       @ZcashFoundation/network-owners @ZcashFoundation/cryptographic-owners
+
+# Network and Async Code
+# To be reviewed by Teor, Janito, or Alfredo.
+/tower-batch/                           @ZcashFoundation/network-owners
+/tower-fallback/                        @ZcashFoundation/network-owners
+/zebra-consensus/                       @ZcashFoundation/network-owners
+/zebra-network/                         @ZcashFoundation/network-owners
+/zebra-state/                           @ZcashFoundation/network-owners
```
Comment (teor2345) [inline .github/CODEOWNERS]: We recently split out this crate:
```suggestion
/zebra-node-services/                   @ZcashFoundation/network-owners
/zebra-state/                           @ZcashFoundation/network-owners
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+/zebra-chain/src/transparent/           @ZcashFoundation/cryptographic-owners
+/zebra-chain/src/history_tree.rs        @ZcashFoundation/cryptographic-owners
+/zebra-chain/src/history_tree/          @ZcashFoundation/cryptographic-owners
+
+# Devops Code
+# To be reviewed by Gustavo, Deirdre, Teor, or Conrado.
+/.github/                               @ZcashFoundation/devops-owners
+/docker/                                @ZcashFoundation/devops-owners
+cloudbuild.yaml                         @ZcashFoundation/devops-owners
+codecov.yml                             @ZcashFoundation/devops-owners
```
Comment (teor2345) [inline .github/CODEOWNERS]: Optional: Did you want to add these files to the devops list?
* .dockerignore
* codecov.yml
* firebase.json
* katex-header.html

I have no opinion either way, just double-checking.

**Comment by teor2345** [comment]
> The teams I created need write permission, and should preferably be under the ZFND team. But I don't have enough permissions to edit the teams with this information.

Someone created the teams already.

I changed the names to "reviewers", so the team names make more sense out of context.
(We already have an "owners" team for crates.io, so I'd like to avoid confusion.)

**Comment by teor2345** [comment]
I also gave the teams direct write permission, which is required by GitHub:
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

**Comment by gustavovalverde** [comment]
@conradoplg can also review this

**Review by conradoplg — APPROVED** [review]
Looks good. I'm a bit worried that the network reviewers will have much more reviews to do, but we can always adjust it later.

**Comment by teor2345** [comment]
> I'm a bit worried that the network reviewers will have much more reviews to do, but we can always adjust it later.

It will depend what we're working on at the time :-)

Since the RPC crate has everyone reviewing, we should be fine for now.

**Pull request merged**